lib/clearance/password_strategies/argon2.rb

```diff
@@ -0,0 +1,41 @@
+module Clearance
+  module PasswordStrategies
+    module Argon2
+      require 'argon2'
+      require 'pry'
+
+      def authenticated?(password)
+        if encrypted_password.present?
+          ::Argon2::Password.verify_password(password, encrypted_password)
+        end
+      end
+
+      def password=(new_password)
+        @password = new_password
+
+        if new_password.present?
+          self.encrypted_password = encrypt(new_password)
+        end
+      end
+
+      private
+
+      # @api private
+      def encrypt(password)
+        t_cost, m_cost = costs
+        argon = ::Argon2::Password.new(t_cost: t_cost, m_cost: m_cost)
+        argon.create(password)
+      end
+
+      # @api private
+      def costs
+        test_environment? ? [1, 4] : [2, 16]
```

Comment:
What do you think about extracting constant for these arrays? 

```
MINIMAL_COST = [1, 4]
DEFAULT_COST = [2, 16]
```

Additionally, we do not use ternaries in this code base as we prefer the explicitness of branching conditionals:

```
if test_environment?
  MINIMAL_COST
else
  DEFAULT_COST
end
```